six/modules/c++/six.sidd/tests/test_byte_swap.cpp

```diff
@@ -118,22 +118,13 @@ void write(const sys::Int16_T* data, bool useStream, bool byteSwap)
     }
 }
 
-void read(const std::string& filename, sys::Int16_T* data)
+void read(const std::string& filename, mem::ScopedArray<sys::Int16_T>& data)
 {
-    six::NITFReadControl reader;
-    reader.load(filename);
-    mem::SharedPtr<const six::Container> container = reader.getContainer();
-    const six::Data* const inData = container->getData(0);
+    std::auto_ptr<six::ReadControl> reader(new six::NITFReadControl());
```

Comment:
Oh yeah... add a `using ReadControl::interleaved` in NITFReadControl... do some Googling to see why that's needed.